src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,669 @@
+    :param Namespace config: Configuration options for pipeline
+    :param str rg_line: RG line for BWA alignment. Default is None.
+    :return: GVCF FileStoreID
+    :rtype: str
+    """
+    config = deepcopy(config)
+    config.uuid = uuid
+    config.url = url
+
+    get_bam = job.wrapJobFn(prepare_bam, uuid, url, url2, config, rg_line=rg_line).encapsulate()
+    job.addChild(get_bam)
+    input_bam = get_bam.rv(0)
+    input_bai = get_bam.rv(1)
+
+    if config.preprocess:
+        preprocess = job.wrapJobFn(run_germline_preprocessing,
+                                   config.cores,
+                                   get_bam.rv(0),
+                                   get_bam.rv(1),
+                                   config.genome_fasta,
+                                   config.genome_dict,
+                                   config.genome_fai,
+                                   config.phase,
+                                   config.mills,
+                                   config.dbsnp,
+                                   xmx=config.xmx).encapsulate()
+        get_bam.addChild(preprocess)
+
+        if config.preprocess_only:
+            output_dir = os.path.join(config.output_dir, uuid)
+            filename = '{}.{}.bam'.format(uuid, config.suffix)
+            output_bam = job.wrapJobFn(upload_or_move_job,
+                                       filename,
+                                       preprocess.rv(0),
+                                       output_dir)
+            preprocess.addChild(output_bam)
+            return preprocess.rv(0), preprocess.rv(1)
         else:
-            samples_file = input_args.manifest
-
-        with open(samples_file, 'r') as f:
-            for line in f.readlines():
-                if not line.isspace() and not line.startswith('#'):
-                    samples.append(line.strip().split('\t'))
-
-
-    for sample in samples:
-        job.addChildJobFn(start, shared_ids, input_args, sample)
+            input_bam = preprocess.rv(0)
+            input_bai = preprocess.rv(1)
+
+    hc_disk = PromisedRequirement(lambda x: 2 * x.size + human2bytes('5G'), input_bam)
+    haplotype_caller = job.wrapJobFn(gatk_haplotype_caller,
+                                     input_bam,
+                                     input_bai,
+                                     config,
+                                     cores=config.cores,
+                                     disk=hc_disk)
+
+    get_bam.addFollowOn(haplotype_caller)
+    output_dir = os.path.join(config.output_dir, uuid)
+    vqsr_name = '{}.raw{}.gvcf'.format(uuid, config.suffix)
+    output_gvcf = job.wrapJobFn(upload_or_move_job,
+                                vqsr_name,
+                                haplotype_caller.rv(),
+                                output_dir)
+    haplotype_caller.addChild(output_gvcf)
+    return haplotype_caller.rv()
+
+
+def download_shared_files(job, config):
```

Comment:
I generally place job functions in a hierarchical order from top to bottom — e.g. since this is called by `startToil` it would be at the top, follow by `reference_preprocessing`